src/backend/distributed/transaction/lock_graph.c

```diff
@@ -737,13 +737,23 @@ AddEdgesForLockWaits(WaitGraph *waitGraph, PGPROC *waitingProc, PROCStack *remai
 	int conflictMask = lockMethodTable->conflictTab[waitingProc->waitLockMode];
 
 	/* iterate through the queue of processes holding the lock */
+#if PG_VERSION_NUM >= PG_VERSION_16
```

Comment:
uf, this is very critical piece of code and now super hard to follow.

Is there a possible refactoring that could make this code easier to follow/review?

Reply:
Let me see what I can do: at the very very least I could add some comments.

Reply:
I made an attempt but it's not a low hanging fruit. Since `AddEdgesForLockWaits` routine is small and it's the only place where `SHMQueueNext` is used, another option is to have two versions of the same routine, one using `SHMQueueNext` and the other using `dlist` model. This will look clean and easy to follow when reading the code.